bzt/engine/engine.py

```diff
@@ -391,7 +391,11 @@ def create_artifacts_dir(self, existing_artifacts=(), merged_config=None):
             artifacts_dir = self.config.get(SETTINGS, force_set=True).get("artifacts-dir", self.ARTIFACTS_DIR)
             self.artifacts_dir = datetime.datetime.now().strftime(artifacts_dir)
 
-        self.artifacts_dir = get_full_path(self.artifacts_dir)
+        envs = self.config.get(SETTINGS, force_set=True).get("env", force_set=True)
+        envs[TAURUS_ARTIFACTS_DIR] = self.artifacts_dir
+        envs = self._expend_env_vars(envs)
+
+        self.artifacts_dir = get_full_path(envs[TAURUS_ARTIFACTS_DIR])
```

Comment:
I don’t like the idea of ​​rebuilding env vars in the method responsible for setting artifacts_dir. I suggest to build 'artifacts_dir' variable only using env state. You can call 'os.path.expandvars' once or while artifacts_dir will not have any variables inside, it depends on what you want to achieve.